tests/integration/p2p.go

```diff
@@ -307,7 +317,7 @@ func configureReplicator(
 				docIDsSyncedToSource[currentdocID] = struct{}{}
 			}
 
-			if action.NodeID.HasValue() && action.NodeID.Value() == cfg.SourceNodeID {
+			if !action.NodeID.HasValue() || action.NodeID.Value() == cfg.SourceNodeID {
```

Comment:
question: Are you sure this is correct?

If `!action.NodeID.HasValue()` then the document will be created synchronously and locally by the test framework in all nodes. Does the P2P system still transmit sync events if they already exist on the node?

todo: If this is correct, can you please add a comment documenting this :)

Reply:
> Does the P2P system still transmit sync events if they already exist on the node?

Yes. The Replicator doesn't know that the target has the document already so it still sends the created document which will cause the target to send a `received push log` event.